packages/storage/exp/index.ts

```diff
@@ -51,8 +51,9 @@ import {
   StorageObserver,
   SettableMetadata,
   UploadMetadata,
-  FullMetadata
-} from '@firebase/storage-types/exp';
+  FullMetadata,
+  UploadTaskSnapshot
+} from './public-types';
```

Comment:
We can directly export them instead of import, then export since `./public-types` is part of the project, so we won't have the typings in the js file issues.

Reply:
Also can we just export every thing from `./public-types`. All types in `./public-types` are supposed to be public/exported.

Reply:
Done. A number of these types are used in this file so I kept those imports.